test/fulfillmentSpec.js

```diff
@@ -177,6 +178,57 @@ describe('GET /fulfillment', function () {
       .expect(this.executionConditionFulfillment)
       .expect(validator.validateFulfillment)
       .end()
+  }
+
+  it('should return a fulfillment', returnFulfillment)
+
+  describe('when hold account is busy', function () {
+    const ARGS_EXPECTED = {
+      '0': 'UPDATE "L_ACCOUNTS" SET "BALANCE" = "BALANCE" + ? WHERE "NAME" = ?',
+      '1': [ -10, 'hold' ]
+    }
+
+    beforeEach(function () {
+      dbFailureMock.timesQueryShouldFail[JSON.stringify(ARGS_EXPECTED)] = 3
+    })
+
+    afterEach(function () {
+      const timesLeft = dbFailureMock.timesQueryShouldFail[JSON.stringify(ARGS_EXPECTED)]
+      dbFailureMock.timesQueryShouldFail[JSON.stringify(ARGS_EXPECTED)] = 0
+      expect(timesLeft).to.equal(0)
+    })
+
+    it('should return a fulfillment', returnFulfillment)
+
+    it('should give up if retrying takes too long', function * () {
+      dbFailureMock.timesQueryShouldFail[JSON.stringify(ARGS_EXPECTED)] = 1000000
+      try {
+        const transfer = this.preparedTransfer
+
+        yield dbHelper.setHoldBalance(10)
+
+        yield this.request()
+          .put(transfer.id + '/fulfillment')
+          .auth('admin', 'admin')
+          .send(this.executionConditionFulfillment)
+          .expect(500)
```

Comment:
@emschwartz if the answer was yes (or if there's anything else that you still want to see improved), then please let me know, and I'll create a follow-up ticket. Merging this now, with all your comments addressed, hopefully.